actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -1264,8 +1278,16 @@ def resources(*resources, &block)
             concerns(options[:concerns]) if options[:concerns]
 
             collection do
-              get  :index if parent_resource.actions.include?(:index)
-              post :create if parent_resource.actions.include?(:create)
+              actions = parent_resource.actions
+              (options[:constraints] ||= {}).merge! parent_resource.collection_param => COLLECTION_PARAM_REGEXP if parent_resource.collection_routing?
+              get    :index, options if actions.include?(:index)
+              post   :create, options if actions.include?(:create)
+              if parent_resource.collection_routing?
+                put    :replace, options if actions.include?(:replace)
+                patch  :update_many, options if actions.include?(:update_many)
+                delete :destroy_many, options if actions.include?(:destroy_many)
+                get    :edit_many, options.merge(as: 'edit') if actions.include?(:edit_many)
+              end
             end
 
             new do
```

Comment:
Regarding your first comment - I actually had to keep `new` below those declarations for the tests to pass. A lot changes in 2 years. I thin I'll come across this while trying to make the tests pass.